cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/httpframeworkhandler/VertxHttpFrameworkHandler.java

```diff
@@ -0,0 +1,161 @@
+
+public class VertxHttpFrameworkHandler implements HttpFrameworkHandler<KafkaCruiseControlConfig> {
+
+    static final String[] HEADERS_TO_TRY = {
+            "X-Forwarded-For",
+            "Proxy-Client-IP",
+            "WL-Proxy-Client-IP",
+            "HTTP_X_FORWARDED_FOR",
+            "HTTP_X_FORWARDED",
+            "HTTP_X_CLUSTER_CLIENT_IP",
+            "HTTP_CLIENT_IP",
+            "HTTP_FORWARDED_FOR",
+            "HTTP_FORWARDED",
+            "HTTP_VIA",
+            "REMOTE_ADDR"
+    };
+
+    protected RoutingContext _context;
+
+    private final CruiseControlHttpSession _session;
+
+    public VertxHttpFrameworkHandler(RoutingContext context) {
+        super();
+        _context = context;
+        _session = new VertxSession(context.session());
+    }
+
+    @Override
+    public String getRequestURL() {
+        return String.format("%s %s", _context.request().method(), _context.request().uri().split("\\?")[0]);
+    }
+
+    @Override
+    public String getUserTaskIdString() {
+        return _context.request().getHeader(USER_TASK_HEADER_NAME);
+    }
+
+    @Override
+    public String getMethod() {
+        return _context.request().method().toString();
+    }
+
+    @Override
+    public String getPathInfo() {
+        return _context.request().uri().split("\\?")[0];
+    }
+
+    @Override
+    public String getClientIdentity() {
+        return getVertxClientIpAddress(_context);
+    }
+
+    @Override
+    public MultiMap getHeaders() {
+        return _context.request().headers();
+    }
+
+    @Override
+    public String getHeader(String header) {
+        return _context.request().getHeader(header);
+    }
+
+    @Override
+    public String getRemoteAddr() {
+        return _context.request().remoteAddress().toString().split(":")[0];
+    }
+
+    @Override
+    public Map<String, String[]> getParameterMap() {
+        return getVertxQueryParamsMap(_context);
+    }
+
+    @Override
+    public String getRequestUri() {
+        return _context.request().uri();
+    }
+
+    @Override
+    public String getParameter(String parameter) {
+        return _context.queryParams().get(parameter);
+    }
+
+    @Override
+    public void writeResponseToOutputStream(int responseCode, boolean json, boolean wantJsonSchema, String responseMessage,
+                                            KafkaCruiseControlConfig config) throws IOException {
+        ResponseUtils.setResponseCode(_context, responseCode, config);
+        _context.response().putHeader("Cruise-Control-Version", KafkaCruiseControl.cruiseControlVersion());
+        _context.response().putHeader("Cruise-Control-Commit_Id", KafkaCruiseControl.cruiseControlCommitId());
+        if (json && wantJsonSchema) {
+            _context.response().putHeader("Cruise-Control-JSON-Schema", getJsonSchema(responseMessage));
+        }
+        _context.response()
+                .end(responseMessage);
```

Comment:
Seems to be copied from `ServletHttpFrameworkHandler`. Please avoid duplicating code.